Library/Homebrew/extend/os/linux/dependency_collector.rb

```diff
@@ -0,0 +1,23 @@
+class DependencyCollector
+  def git_dep_if_needed(tags)
+    Dependency.new("git", tags)
+  end
+
+  def cvs_dep_if_needed(tags)
+    Dependency.new("cvs", tags)
+  end
+
+  def xz_dep_if_needed(tags)
+    Dependency.new("xz", tags)
+  end
+
+  def ld64_dep_if_needed(*); end
+
+  def zip_dep_if_needed(tags)
+    Dependency.new("zip", tags)
+  end
+
+  def bzip2_dep_if_needed(tags)
+    Dependency.new("bzip2", tags)
+  end
+end
```

Comment:
Remove this file.

Reply:
Same as above: had to add it back to override `xz_dep_if_needed` (https://github.com/Homebrew/brew/pull/3813/commits/14d7a7a08c9aab7db14ea03f59ec85899e5e3ba6#r169467375)

Reply:
Don't override `xz_dep_if_needed` for Linux. This implementation ought to be the generic implementation.

Reply:
Does `xz` exist on all Linux systems? If so - sure. Otherwise, it would make sense to keep it conditional.